src/components/Question.jsx

```diff
@@ -463,22 +505,26 @@ export default class Question extends React.Component {
     return question[Constants.HAS_UNIT] ? <div className="has-unit-label">{question[Constants.HAS_UNIT]}</div> : null;
   }
 
-  renderSubQuestions() {
+  renderSubQuestions(classname) {
     const children = [];
     const subQuestions = this._getSubQuestions();
+    const debugMode = this.context.options.debugMode;
+    const startingQuestionId = this.context.options.startingQuestionId;
 
     for (let i = 0; i < subQuestions.length; i++) {
-
       let question = subQuestions[i];
+      let questionId = question['@id'];
       let component = this.context.mapComponent(question, Question);
-
-      let element = React.createElement(component, {
-        key: 'sub-question-' + i,
-        question: question,
-        onChange: this.onSubQuestionChange,
-        index: i
-      });
-
+      let element = null;
+
+      if (debugMode || classname !== "show-irrelevant" || (!debugMode && questionId === startingQuestionId)) {
```

Comment:
`questionId === startingQuestionId`

IMHO, Above we implemented this functionality differently, don't see reason why it should be different.